joern-cli/frontends/jimple2cpg/src/main/scala/io/joern/jimple2cpg/Jimple2Cpg.scala

```diff
@@ -116,6 +113,26 @@ class Jimple2Cpg {
     }
   }
 
+  private def loadSourceFiles(
+      sourceCodePath: String,
+      sourceFileExtensions: Set[String],
+      archives: List[String]
+  ) = {
+    (archives
+      .map(new ZipFile(_))
+      .flatMap(x => {
+        unzipArchive(x, sourceCodePath) match {
+          case Failure(e) =>
+            logger.error(s"Error extracting files from archive at ${x.getName}", e); null
```

Comment:
`null` is evil, please don't use it, to avoid NullPointers. 
In Scala, we have Try, which you already use in `unzipArchive`. You can decide if you want to return a `Try` here or just deal with the happy path, in which case you should throw an exception in this line.